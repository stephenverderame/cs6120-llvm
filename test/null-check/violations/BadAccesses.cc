#include <cassert>
#include <cstdio>
#include <cstdlib>
// NOLINTBEGIN

void access(int (&foo)[100])
{
    unsigned rnd = rand();
    int srnd = rand();
    int a = rnd % 7u;
    int b = (rnd % 10u) + 6;
    int c = rnd % 12u;
    foo[b - c] = 0;
    foo[a * a * a] = 25;
    foo[a * 15 - b] = 10;
    foo[a * a + b * b] = 5;
    foo[a + a + b] = 0;
}

int main(int argc, const char** argv)
{
    int foo[100];
    access(foo);
}
// NOLINTEND