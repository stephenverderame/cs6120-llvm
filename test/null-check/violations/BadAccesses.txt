[31mSafety Violation[00m: Use of potentially null pointer in store at BadAccesses.cc:13
Because an index could not be proven to be in range, the following counterexample was found:
	[1mb[0m = 7
	[1mc[0m = 11
	[1mrnd[0m = 71
[31mSafety Violation[00m: Use of potentially null pointer in store at BadAccesses.cc:14
Because an index could not be proven to be in range, the following counterexample was found:
	[1ma[0m = 5
	[1mrnd[0m = 5
[31mSafety Violation[00m: Use of potentially null pointer in store at BadAccesses.cc:15
Because an index could not be proven to be in range, the following counterexample was found:
	[1ma[0m = 0
	[1mb[0m = 6
	[1mrnd[0m = 0
[31mSafety Violation[00m: Use of potentially null pointer in store at BadAccesses.cc:16
Because an index could not be proven to be in range, the following counterexample was found:
	[1ma[0m = 5
	[1mb[0m = 11
	[1mrnd[0m = 75
