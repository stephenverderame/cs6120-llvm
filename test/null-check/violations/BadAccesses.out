[31mSafety Violation[00m: Use of potentially null pointer in store at BadAccesses.cc:13
[31mSafety Violation[00m: Use of potentially null pointer in store at BadAccesses.cc:14
[31mSafety Violation[00m: Use of potentially null pointer in store at BadAccesses.cc:15
[31mSafety Violation[00m: Use of potentially null pointer in store at BadAccesses.cc:16
